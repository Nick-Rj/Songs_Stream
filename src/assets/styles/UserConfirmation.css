.UserConfirmation {
  background-color: #2789b0b9;
  height: 100vh;
  width: 100vw;
}

.UserConfirmation .profile-card {
  margin-top: 7rem;
}

.UserConfirmation h1 {
  padding-top: 2em;
  /* color: #ecd156; */
  padding-bottom: 1em;
  font-weight: lighter;
}

.UserConfirmation h3 {
  /* color: #ecd156; */
  color: rgb(56, 219, 105);
  padding-bottom: 0.5em;
}

.UserConfirmation p {
  /* color: #ecd156; */
  font-size: 1.3em;
  color: rgb(56, 219, 105);
  font-weight: bold;
}

.UserConfirmation Button {
  margin-top: 1em;
  margin-left: 1em;
}

@media screen and (max-width: 768px) {
  .UserConfirmation h1 {
    font-size: 1.4rem;
    margin-top: 2.5rem;
  }

  .UserConfirmation p {
    font-size: small;
  }
}
