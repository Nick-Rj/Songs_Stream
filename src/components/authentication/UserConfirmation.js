import React, { useEffect, useState } from "react";
import { Button, Container, Row, Col, Card } from "react-bootstrap";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { clearCurrentUser } from "../../redux/store";
import "../../assets/styles/UserConfirmation.css";
import styled, { keyframes } from "styled-components";
import { fadeIn } from "react-animations";

const AnimateRegisterConfirm = styled.div`
  animation: 1.5s ${keyframes`${fadeIn}`};
`;

function UserConfirmation({ userData, clearCurrentUser }) {
  const variants = [
    "Primary",
    "Secondary",
    "Success",
    "Danger",
    "Info",
    "Light",
    "Dark",
  ];

  const random = Math.floor(Math.random() * variants.length);
  const selectedVariant = variants[random];

  const [user, setUser] = useState(userData.user);

  return (
    <div className="UserConfirmation">
      <AnimateRegisterConfirm>
        <Container>
          <Row>
            <Col xs={1}></Col>
            <Col>
              {" "}
              <Card
                bg={selectedVariant.toLowerCase()}
                text={
                  selectedVariant.toLowerCase() === "light" ? "dark" : "white"
                }
                className="mb-2 profile-card"
              >
                <Card.Header>User Profile</Card.Header>
                <Card.Body>
                  <Card.Title>Here are your details! </Card.Title>
                  <Card.Text>
                    <Container>
                      <Row>
                        <Col>
                          <p>First Name: {userData.user.firstName}</p>
                          <p>Last Name: {userData.user.lastName}</p>
                          <p>Location: {userData.user.location}</p>
                          <p>Phone: {userData.user.phone}</p>
                          <p>Email: {userData.user.email}</p>

                          <Link to="/login">
                            <Button
                              variant={
                                selectedVariant.toLowerCase() === "success"
                                  ? "danger"
                                  : "success"
                              }
                            >
                              Login
                            </Button>
                          </Link>
                          <Link to="/songsLibrary">
                            <Button
                              variant={
                                selectedVariant.toLowerCase() === "warning"
                                  ? "light"
                                  : "warning"
                              }
                              onClick={clearCurrentUser}
                            >
                              Go to Songs Library
                            </Button>
                          </Link>
                        </Col>
                      </Row>
                    </Container>
                  </Card.Text>
                </Card.Body>
              </Card>
            </Col>
            <Col xs={1}></Col>
          </Row>
        </Container>
      </AnimateRegisterConfirm>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    userData: state.user,
  };
};

export default connect(mapStateToProps, { clearCurrentUser })(UserConfirmation);
