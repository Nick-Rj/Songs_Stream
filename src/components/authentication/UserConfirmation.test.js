import { shallow } from "enzyme";
import { expect, it } from "@jest/globals";

import UserConfirmation from "./UserConfirmation";

import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({
  userState: {
    loading: false,
    users: [],
    user: {},
    error: "",
    isLoggedIn: false,
  },
  songData: { loading: false, songs: [], error: "" },
  playlistData: { loading: false, playlists: [], error: "" },
});

it("rendering full DOM for UserConfirmation Component", () => {
  const wrapper = shallow(
    <Provider store={store}>
      <UserConfirmation />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
