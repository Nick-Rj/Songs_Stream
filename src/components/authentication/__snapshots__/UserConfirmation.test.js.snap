// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering full DOM for UserConfirmation Component 1`] = `ShallowWrapper {}`;
